src/modules/registrypreview/RegistryPreviewUI/RegistryPreviewXAML/App.xaml.cs

```diff
@@ -52,6 +53,20 @@ protected override void OnLaunched(Microsoft.UI.Xaml.LaunchActivatedEventArgs ar
                     }
                 }
             }
+            else if (activatedArgs.Kind == ExtendedActivationKind.Protocol)
+            {
+                // When the app is the default handler for REG files and the filename has non-ASCII characters, the app gets activated by Protocol
+                AppFilename = string.Empty;
+                if (activatedArgs.Data != null)
+                {
+                    IProtocolActivatedEventArgs eventArgs = (IProtocolActivatedEventArgs)activatedArgs.Data;
+                    if (eventArgs.Uri.AbsoluteUri.Length > 0)
+                    {
+                        AppFilename = eventArgs.Uri.Query.Replace("?ContractId=Windows.File&Verb=open&File=", string.Empty);
+                        AppFilename = HttpUtility.UrlDecode(AppFilename);
+                    }
+                }
+            }
```

Comment:
That this if is required at all needs some explanation from someone in the UWP or WinUI 3 stack.